security/pkg/nodeagent/caclient/client.go

```diff
@@ -45,13 +45,12 @@ type configMap interface {
 }
 
 // NewCAClient create an CA client.
-func NewCAClient(endpoint, caProviderName string, tlsFlag bool, tlsRootCert []byte, vaultAddr, vaultRole,
-	vaultAuthPath, vaultSignCsrPath string) (caClientInterface.Client, error) {
+func NewCAClient(endpoint, caProviderName string, tlsFlag bool, config string) (caClientInterface.Client, error) {
```

Comment:
The parameter "config" is a string with the following 7 Vault parameters delimited by ";".  It is error-prone to use a long string to represent 7 parameters and each parameter may contain ";" itself, making the parsing problematic. Therefore, I suggest using a structure to represent these 7 Vault related parameters, e.g., 
type VaultConfig struct {
   vaultAddr string
   tlsRootCertPath string
   ....
} 

	vaultAddr := params[0]
	tlsRootCertPath := params[1]
	jwtPath := params[2]
	vaultLoginRole := params[3]
	vaultLoginPath := params[4]
	vaultSignCsrPath := params[5]
	vaultCACertPath := params[6]

Reply:
This string will be user facing, so we need to make sure it can be expressed by a string. I picked ";" because it won't present in these fields.

Reply:
If this string is user facing, with the current implementation in this PR, a user need to assemble a long string with 7 parameters. Will it be easier for a user to config 7 parameters separately? The pros of configuring 7 parameters separately include:
1) A user does not need to assemble 7 parameters together. 
2) The PR does not need to parse a long string into 7 parameters.

Reply:
We want to avoid adding Vault-specific code outside of the function. TBH, I don't worry that much about the format of the parameters because this isn't a frequently used config.
But I'm thinking having vault/client.go directly load the ENV inside it - the code outside don't need to care about Vault ENVs.